### PR TITLE
Service account manager impl

### PR DESCRIPTION
**Pull request by MaciejMierzwa**
### Description
Hi,
Here's draft PR of changes for task [6900](https://github.com/opensearch-project/OpenSearch/issues/6900). ServiceAccountManagerImpl implements core-side ServiceAccountManager interface and is managed by class implementing IdentityPlugin. Please let  me know if it's a correct approach. 

### Issues Resolved
[6900](https://github.com/opensearch-project/OpenSearch/issues/6900)

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/security/identity/AccountServiceIdentityPlugin.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.User;
+import org.opensearch.identity.UserProvider;
+import org.opensearch.plugins.IdentityPlugin;
+import org.opensearch.security.user.UserService;
+
+import java.io.IOException;
+import java.util.List;
+
+public class AccountServiceIdentityPlugin implements IdentityPlugin {
```
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/identity/AccountServiceIdentityPlugin.java:22]: thoughts on renaming this to: `ServiceAccountPlugin`

File: src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java [hunk 2]
```diff
@@ -338,7 +338,7 @@ protected void addAdditionalContextValues(final String action, final TransportRe
         String extensionUniqueId = getThreadContext().getHeader("extension_unique_id");
         if (extensionUniqueId != null) {
             ExtensionsManager extManager = OpenSearchSecurityPlugin.GuiceHolder.getExtensionsManager();
-            if (extManager.lookupExtensionSettingsById(extensionUniqueId).isPresent()) {
+            if (extManager.lookupInitializedExtensionById(extensionUniqueId).isPresent()) {
```
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java:341]: this change should be reverted as lookupExtensionSettingsById() is the correct method call.

**Comment by peternied** [comment]
Looks like this pull request has stalled for a little bit @MaciejMierzwa thoughts on picking this back up?

**Comment by MaciejMierzwa** [comment]
Hi, sorry for the confusion @peternied, this is obsolete now. Closing the PR

**Pull request closed without merging**